pkgs/by-name/mr/mricron/package.nix

```diff
@@ -0,0 +1,61 @@
+{ lib, stdenv, fetchurl, makeWrapper, unzip, autoPatchelfHook, makeDesktopItem, copyDesktopItems
+, gtk2, glib, gdk-pixbuf, pango, cairo, atk
+, freetype, fontconfig, xorg, zlib
+}:
+
+let
+  pkgDescription = "A application to display NIfTI medical imaging data.";
+in
+
+stdenv.mkDerivation rec {
+  pname = "mricron";
+  version = "1.0.20190902";
+
+  src = fetchurl {
+    url = "https://github.com/neurolabusc/MRIcron/releases/download/v${version}/MRIcron_linux.zip";
+    sha256 = "sha256-C155u9dvYEyWRfTv3KNQFI6aMWIAjgvdSIqMuYVIOQA=";
+  };
+
+  nativeBuildInputs = [ makeWrapper unzip autoPatchelfHook copyDesktopItems ];
+  buildInputs = [ gtk2 glib gdk-pixbuf pango cairo atk freetype fontconfig xorg.libX11 zlib ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/${pname}
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+
+    install -Dm555 ./MRIcron $out/bin/${pname}
+    install -Dm444 -t $out/share/icons/hicolor/scalable/apps/ ./Resources/mricron.svg
+    cp -r ./Resources/* $out/share/${pname}/
+
+    runHook autoPatchelfHook $out/bin/${pname}
```

Comment:
wouldn't `autoPatchelf $out/bin/${pname}` do the trick? Also if you only want to patch `$out/bin/${pname}` you should add `dontAutoPatchelf = true;`

Reply:
Is it rigth to have to install -Dm 777  for autopatchelf to work?    the run hook worked with 555, but now i need write privilege as well.

Reply:
you can completely remove the line `runHook autoPatchelfHook $out/bin/${pname}` The hook gets called automatically and since `mricron` is the only binary in `bin` it is the only one which gets patched